fpdf/fpdf.py

```diff
@@ -704,14 +713,26 @@ def add_page(
             # close page
             self._endpage()
 
-        # Start new page
-        self._beginpage(
-            orientation,
-            format,
-            same,
-            duration or self.page_duration,
-            transition or self.page_transition,
-        )
+        if self._has_next_page():
```

Comment:
Instead of duplicating the call to `_beginpage`,
you could do this:
```python
has_next_page = self.pages_count > self.page
self._beginpage(
    orientation,
    format,
    same,
    duration or self.page_duration,
    transition or self.page_transition,
    new_page=not has_next_page,
)
```